srml/contract/src/account_db.rs

```diff
@@ -46,50 +45,11 @@ impl<T: Trait> Default for ChangeEntry<T> {
 
 pub type ChangeSet<T> = BTreeMap<<T as system::Trait>::AccountId, ChangeEntry<T>>;
 
-#[derive(Clone, Default)]
-pub struct AccountTrieIdMapping<A: Ord> {
-	to_account: BTreeMap<TrieId, A>,
-	to_key: BTreeMap<A, TrieId>,
-	// this lock is related to the way overlaydb stack
-	// if set it must be unset at the lower level
-	lock: bool,
-}
-
-impl<A: Clone + Ord> AccountTrieIdMapping<A> {
-
-	pub fn new() -> Self {
-		AccountTrieIdMapping {
-			to_account: BTreeMap::new(),
-			to_key: BTreeMap::new(),
-			lock: false,
-		}
-	}
-
-	pub fn lock(&mut self) {
-		self.lock = true;
-	}
-	pub fn unlock(&mut self) {
-		self.lock = false;
-	}
-	pub fn insert(&mut self, account: A, ks: TrieId) {
-		self.to_account.insert(ks.clone(), account.clone());
-		self.to_key.insert(account, ks);
-	}
-	pub fn get_trieid(&self, account: &A) -> Option<&TrieId> {
-		if self.lock { return None }
-		self.to_key.get(account)
-	}
-	pub fn get_account(&self, ks: &TrieId) -> Option<&A> {
-		if self.lock { return None }
-		self.to_account.get(ks)
-	}
-
-}
-
 pub trait AccountDb<T: Trait> {
-	fn get_account_info(&self, account: &T::AccountId) -> Option<AccountInfo>;
-	fn get_or_create_trieid(&self, account: &T::AccountId) -> TrieId;
-	fn get_storage(&self, trie_id: &TrieId, location: &[u8]) -> Option<Vec<u8>>;
+	fn get_final_trie_id(&self, account: &T::AccountId) -> Option<TrieId>;
```

Comment:
`get_code` and `get_balance` are pretty self-explanatory methods. `get_storage` used to be a self-explanatory until we introduced complications with `trie_id`. `get_final_trie_id` can be difficult to understand from the first sight. Especially for a new comer.

So, can we have a few lines of docs on what exactly this method should return and in what circumstances? In which situations `Option` can be `None` and etc.

These kind of comments will help us onboard new contributors, ease audit process a bit and in some cases it will help us understand our code better : )

Reply:
Yes I improved it a bit